core/src/labels/labels.cpp

```diff
@@ -104,6 +105,54 @@ void Labels::update(const View& _view, float _dt, const std::vector<std::unique_
         }
     }
 
+    /// Mark labels to skip transitions
+
+    for (const auto& t0 : _tiles) {
```

Comment:
Yeah right! We should find a way to only trigger this loop when a tile transition happens.